### PR TITLE
Refactor large tests datalake

### PR DESCRIPTION
**Pull request by kasobol-msft**
Remaining part of https://github.com/Azure/azure-sdk-for-java/issues/12709

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/LargeFileTest.groovy [hunk 1]
```diff
@@ -8,143 +8,101 @@ import com.azure.core.http.HttpResponse
...
+import spock.lang.ResourceLock
 import spock.lang.Unroll
 
 import java.nio.ByteBuffer
 import java.util.concurrent.atomic.AtomicLong
-import java.util.function.BiFunction
 
+@LiveOnly
+@ResourceLock("LargeFileTest")
+@IgnoreIf({ TestEnvironment.getInstance().httpClientType == TestHttpClientType.OK_HTTP || SystemUtils.IS_OS_MAC }) // https://github.com/Azure/azure-sdk-for-java/issues/23221
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/LargeFileTest.groovy:27]: - mac agents are not collocated with storage account, so disabling there
- okhttp doesn't like payloads larger than int.max...

**Review by alzimmermsft — APPROVED** [review]

**Review by rickle-msft — APPROVED** [review]

File: sdk/storage/azure-storage-blob/pom.xml [hunk 2]
```diff
@@ -239,7 +239,7 @@
             <version>3.0.0-M3</version> <!-- {x-version-update;org.apache.maven.plugins:maven-surefire-plugin;external_dependency} -->
             <configuration>
               <argLine>
-                -Xmx6g -Xms6g
+                -Xmx6g
               </argLine>
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-blob/pom.xml:243]: Ci didn't like that

**Pull request merged**